### PR TITLE
fix(test): fix some flaky e2e source tests

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- `temporary_kafka_batch.slt`: In #18894, we moved data generation from ci script (generate before tests) to `slt` (generate inline), which may cause the test fail.
- `glue.slt`: Alter license may not take effect immediately.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by stdrc — APPROVED** [review]

File: e2e_test/source_inline/kafka/avro/glue.slt [hunk 1]
```diff
@@ -14,6 +14,8 @@ EOF
 statement ok
 ALTER SYSTEM SET license_key TO '';
 
+sleep 1s
```
Comment (BugenZhao) [inline e2e_test/source_inline/kafka/avro/glue.slt:17]: There are also many other occurrences which may explode in the coming days, I suppose. 🤣

It's indeed possible to return the `ALTER SYSTEM` statement only after the notification has been broadcast to all nodes in the cluster, but the process of receiving and handling the notification remains asynchronous. So I'm afraid we are not able to completely eliminate the flakiness.

File: e2e_test/source_inline/kafka/temporary_kafka_batch.slt [hunk 2]
```diff
@@ -71,6 +71,12 @@ t
 t
 t
 
+# ensure now() is larger
+sleep 1s
```
Comment (BugenZhao) [inline e2e_test/source_inline/kafka/temporary_kafka_batch.slt:75]: The value of `now()` is updated if there's a new hummock snapshot synchronized to the frontend. By `FLUSH` we are sure there's a new snapshot generated, but the notification still remains asynchronous.

**Review by BugenZhao — APPROVED** [review]

File: e2e_test/source_inline/kafka/temporary_kafka_batch.slt (hunk 2 above)
Comment (xxchan) [reply e2e_test/source_inline/kafka/temporary_kafka_batch.slt:75]: 🧐 I had thought after `FLUSH` we will definitely go to next epoch

Comment (BugenZhao) [reply e2e_test/source_inline/kafka/temporary_kafka_batch.slt:75]: > I had thought after `FLUSH` we will definitely go to next epoch

Ah you're right. I'm a little dumb.

https://github.com/risingwavelabs/risingwave/blob/b3655662fd366845a99caaf2b3e95dbb5bbffbe1/src/frontend/src/handler/flush.rs#L31-L37

File: e2e_test/source_inline/kafka/avro/glue.slt (hunk 1 above)
Comment (yuhao-su) [reply e2e_test/source_inline/kafka/avro/glue.slt:17]: > handling the notification remains asynchronous

Is it possible we make it a synchronous process on meta by altering the parameter before repling to fe?

Comment (BugenZhao) [reply e2e_test/source_inline/kafka/avro/glue.slt:17]: It is synchronous on meta node, but the following check is performed on the frontend IIUC.

**Pull request merged**